src/server/repositories/product.repository.js

```diff
@@ -8,11 +8,18 @@ export const getOneById = async (id) => {
       id,
     },
   });
-  return product[0];
+  if (product[0]) {
+    return product[0];
+  }
+  return false;
```

Comment:
this is not a good practice to mix return types. In this case, you will return an object when the product exists and a boolean if it does not exist. It's better to return `null` instead of `false` to make result more type consistent.